src/test/suite/completion.test.ts

```diff
@@ -377,6 +351,29 @@ describe("Should do completion", () => {
       "const d"
     );
   });
+  it("should cancel running debounce request upon new request", async () => {
+    mockGetDebounceConfig(LONG_DEBOUNCE_VALUE);
+    mockAutocomplete(
+      requestResponseItems,
+      anAutocompleteResponse("d", "data"),
+      anAutocompleteResponse("da", "data1")
+    );
+
+    await openADocWith("const ", "text");
+
+    await vscode.commands.executeCommand("type", {
+      text: "d",
+    });
+    await sleep(200);
+    await vscode.commands.executeCommand("type", {
+      text: "a",
+    });
+    await emulationUserInteraction();
+
+    verify(
+      stdinMock.write(new SimpleAutocompleteRequestMatcher("const d"), "utf8")
```

Comment:
should have been `"const da"`, no? that's the whole point of the debounce, to send only the last request

Reply:
no, the debounce is on the "rendering" (== before the suggestion is returned to the provider) 
the binary is queried at most twice and at least once, at the following points : 
 - the completion provider is queried for competition
 - after debouncing - in order to provide the latest cached results
 if between the two calls a cancelation was issued the second request will be canceled